components/pedigree/resources/src/main/resources/pedigree/view/person.js

```diff
@@ -1571,5 +1565,30 @@ define([
 
     //ATTACHES CHILDLESS BEHAVIOR METHODS TO THIS CLASS
     Person.addMethods(ChildlessBehavior);
+
+    /**
+     * Returns the index of the given gene in the given gene array
+     * (assuming gene array format is the one used internally by Person),
+     * or -1 if not present.
+     *
+     * @param {String} geneID
+     * @param {Array} geneArray
+     */
+    Person.getGeneIndex = function(geneID, geneArray)
+    {
+        for (var i = 0; i < geneArray.length; i++) {
+            if (geneArray[i].hasOwnProperty("id")) {
```

Comment:
because the other "else" should happen only if `!geneArray[i].hasOwnProperty("id")`, but not when `geneArray[i].hasOwnProperty("id")` and `geneArray[i].id.toUpperCase() != geneID.toUpperCase()`